src/west/manifest.py

```diff
@@ -776,7 +778,7 @@ def __str__(self):
         path_repr = repr(self.abspath or self.path)
         return f'<Project {self.name} ({path_repr}) at {self.revision}>'
 
-    def __init__(self, name: str, url: str,
+    def __init__(self, name: str, description: Optional[str], url: str,
```

Comment:
The commit message description is empty and I'm not sure why this is being added as an arg instead of as a kwarg. If we need to add it as an arg like this for a good reason, we can deal with the API change (it's not a big deal as I don't expect "anyone" is making their own Project objects). But if we can have this as `description: Optional[str] = None`, thus keeping compatibility, why not do that?

Reply:
> If we need to add it as an arg like this for a good reason, we can deal with the API change

There isn't a good reason, this was just a first quick pass to gather feedback like the one I got from you now. I will change to become a kwarg. Thanks!